### PR TITLE
chore: update make_url utility with fallback

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

the original `make_url` utility used to handle some unexpected `path` arguments passed
where the update version did not and this was letting some misusage of `make_url` to work which updating
the utility revealed.

- provide and log more info when requests fail to match an api handler
- revert default make_url to the old implementation where it was forgiving of bad path parameters in some cases.
- add a test for accessing notebooks through proxy
- explicitly reject paths parameters set as absolute 

## Commentary (optional)

to avoid having to audit every usage of make_url and requests throughout the system we put back the original implementation and provide the new make_url as a new option to gradually adopt.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/tests/common/test_api.py [hunk 1]
```diff
@@ -19,10 +47,13 @@ def gen_make_url_cases() -> List[Case]:
         Case("http://localhost:8080/", "api/v1/experiments/", f"{host}/{path}/"),
         Case("http://localhost:8080/proxied/", "", f"{host}/proxied/"),
         Case("http://localhost:8080/proxied", "", f"{host}/proxied"),
-        Case("http://localhost:8080/proxied", "/api/v1/experiments", f"{host}/proxied/{path}"),
-        Case("http://localhost:8080/proxied", "api/v1/experiments/", f"{host}/proxied/{path}/"),
-        Case("http://localhost:8080/proxied/", "/api/v1/experiments", f"{host}/proxied/{path}"),
-        Case("http://localhost:8080/proxied/", "api/v1/experiments/", f"{host}/proxied/{path}/"),
+        # Case("http://localhost:8080/proxied", f"/{path}", f"{host}/proxied/{path}"),
+        # Case("http://localhost:8080/proxied", f"{path}/", f"{host}/proxied/{path}/"),
+        # Case("http://localhost:8080/proxied/", f"/{path}", f"{host}/proxied/{path}"),
```
Comment (eecsliu) [inline harness/tests/common/test_api.py:52]: is there a reason to keep them commented?

File: e2e_tests/tests/command/test_notebook.py [hunk 2]
```diff
@@ -17,3 +21,31 @@ def test_basic_notebook_start_and_kill() -> None:
...
+        NTSC_Kind(typ),
+        created_id,
+        lambda s: s == bindings.taskv1State.RUNNING,
+        timeout=300,
+    )
+    deets = get_ntsc_details(session, typ, created_id)
+    assert deets.state == bindings.taskv1State.RUNNING, f"{typ} should be running"
+    err = api.task_is_ready(determined_test_session(ADMIN_CREDENTIALS), created_id)
+    assert err is None, f"{typ} should be ready {err}"
+    print(deets)
```
Comment (eecsliu) [inline e2e_tests/tests/command/test_notebook.py:47]: nit: can we add some a prefix so readers will know what this is? like `print("ntsc details:", deets)`

**Review by eecsliu — APPROVED** [review]
Looks good! Minor nits

File: e2e_tests/tests/command/test_notebook.py (hunk 2 above)
Comment (hamidzr) [reply e2e_tests/tests/command/test_notebook.py:47]: I didn't mean to leave this print there.

File: harness/tests/common/test_api.py (hunk 1 above)
Comment (hamidzr) [reply harness/tests/common/test_api.py:52]: to show case this doesn't support these. I need to update the branch to make the diffs a bit more clear

File: master/internal/core.go [hunk 3]
```diff
@@ -1173,7 +1173,10 @@ func (m *Master) Run(ctx context.Context) error {
 	// Catch-all for requests not matched by any above handler
 	// echo does not set the response error on the context if no handler is matched
 	m.echo.Any("/*", func(c echo.Context) error {
-		return echo.ErrNotFound
+		id := fmt.Sprintf("%s %s", c.Request().Method, c.Request().URL.Path)
```
Comment (hamidzr) [inline master/internal/core.go:1176]: is it too much to log these? maybe at debug level?

Comment (eecsliu) [reply master/internal/core.go:1176]: yeah I would log these at debug level - I think the error message is clear enough to the user that the endpoint doesn't exist

**Pull request merged**